src/html5-qrcode-scanner.ts

```diff
@@ -403,9 +443,16 @@ export class Html5QrcodeScanner {
     }
 
     private setupInitialDashboard(dashboard: HTMLElement) {
+        const $this = this;
         this.createSection(dashboard);
         this.createSectionControlPanel();
-        this.createSectionSwap();
+        if(this.config.supportedScanTypes?.includes(
+            Html5QrcodeScanType.SCAN_TYPE_CAMERA) &&
+            this.config.supportedScanTypes?.includes(
+                Html5QrcodeScanType.SCAN_TYPE_FILE)){
+            this.createSectionSwap();
+        }
```

Comment:
This is nice, how about we abstract some of the logic outside to a separate class.  (better separation of concerns)

```ts
class ScanTypeSelector {

    public constructor(supportedScanTypes?: Array<Html5QrcodeScanType>) {}

    public getDefaultScanType(): Html5QrcodeScanType { /* impl */ }

    private validateAndReturnScanTypes(supportedScanTypes?: Array<Html5QrcodeScanType>)
        : Array<Html5QrcodeScanType> { /** impl. */ } 
}
```

This can be separately well tested and used directly here. (We can discuss more offline).

Reply:
I will work on this soon.